src/App.tsx

```diff
@@ -1,116 +1,81 @@
 import React from 'react';
-import style from './App.module.scss';
-import Header from './components/Header/Header';
-import CardItems from './components/CardItems/CardItems';
-import { URL } from './models/enums';
+import { ROUTER_PATHS, URL } from './models/enums';
 import { ShowData } from './models/interfaces';
-import { MdReportGmailerrorred } from 'react-icons/md';
-import ErrorBoundary from './components/ErrorBoundary/ErrorBoundary';
-
-interface AppProps {
-  shows: ShowData[];
-  isLoading: boolean;
-  error: Error | null | unknown;
-  currentPage: number;
-  itemsPerPage: number;
-  isMoreShows: boolean;
-  searchQuery: string;
-  isShowMoreButtonDisable: boolean;
-  isClickedErrorButton: boolean;
-}
-class App extends React.Component<object, AppProps> {
-  constructor(props: AppProps) {
-    super(props);
-    this.state = {
-      shows: [],
-      isLoading: false,
-      error: null,
-      currentPage: 0,
-      itemsPerPage: 20,
-      isMoreShows: false,
-      searchQuery: '',
-      isShowMoreButtonDisable: false,
-      isClickedErrorButton: false,
-    };
-
-    this.loadShows = this.loadShows.bind(this);
-    this.showMoreShows = this.showMoreShows.bind(this);
-    this.handleSearch = this.handleSearch.bind(this);
-    this.enableButton = this.enableButton.bind(this);
-    this.throwError = this.throwError.bind(this);
-  }
-
-  async componentDidMount() {
-    this.loadShows();
-    this.enableButton();
-  }
-
-  async loadShows() {
-    const { currentPage, isLoading } = this.state;
-
+import { Route, Routes, useNavigate } from 'react-router-dom';
+import Layout from './components/Layout/Layout';
+import MainPage from './components/pages/MainPage/MainPage';
+import NotFound from './components/pages/NotFound/NotFound';
+import DetailsPage from './components/pages/DetailsPage/DetailsPage';
+
+const App: React.FC = () => {
+  const [shows, setShows] = React.useState<ShowData[]>([]);
+  const [isLoading, setIsLoading] = React.useState<boolean>(false);
+  const [error, setError] = React.useState<Error | null | unknown>(null);
+  const [itemsPerPage] = React.useState(20);
+  const [searchQuery, setSearchQuery] = React.useState<string>('');
+  const [isShowMoreButtonDisable, setIsShowMoreButtonDisable] = React.useState<boolean>(false);
+  const [currentPage, setCurrentPage] = React.useState(0);
```

Comment:
specify the type

Reply:
these values are already stored in redux, I added types for other values, defined by useState hook in module 04